serial-settings/src/lib.rs

```diff
@@ -0,0 +1,374 @@
+                            "{path}: {}",
+                            core::str::from_utf8(&context.buffer[..len])
+                                .unwrap()
+                        )
+                        .unwrap(),
+                    }
+
+                    match defaults.get_json(&path, context.buffer) {
+                        Err(e) => writeln!(
+                            context.platform.interface_mut(),
+                            "[default serialization error: {e}]"
+                        ),
+                        Ok(len) => writeln!(
+                            context.platform.interface_mut(),
+                            " [default: {}]",
+                            core::str::from_utf8(&context.buffer[..len])
+                                .unwrap()
+                        ),
+                    }
+                }
+            }
+            .unwrap()
+        }
+    }
+
+    fn handle_clear(
+        _menu: &menu::Menu<Self>,
+        _item: &menu::Item<Self>,
+        _args: &[&str],
+        context: &mut Self,
+    ) {
+        context.platform.settings_mut().reset();
+        match context.platform.save(context.buffer) {
+            Ok(_) => {
+                writeln!(context, "Settings cleared to defaults and saved.")
+            }
+            Err(e) => {
+                writeln!(context, "Failed to clear settings: {e:?}")
+            }
+        }
+        .unwrap();
+    }
+
+    fn handle_get(
+        _menu: &menu::Menu<Self>,
+        item: &menu::Item<Self>,
+        args: &[&str],
+        context: &mut Self,
+    ) {
+        let key = menu::argument_finder(item, args, "item").unwrap().unwrap();
+        match context.platform.settings().get_json(key, context.buffer) {
+            Err(e) => {
+                writeln!(
+                    context.platform.interface_mut(),
+                    "Failed to read {key}: {e}"
+                )
+            }
+            Ok(len) => {
+                writeln!(
+                    context.platform.interface_mut(),
+                    "{key}: {}",
+                    core::str::from_utf8(&context.buffer[..len]).unwrap()
+                )
+            }
+        }
+        .unwrap();
+    }
+
+    fn handle_set(
+        _menu: &menu::Menu<Self>,
+        item: &menu::Item<Self>,
+        args: &[&str],
+        context: &mut Self,
+    ) {
+        let key = menu::argument_finder(item, args, "item").unwrap().unwrap();
+        let value =
+            menu::argument_finder(item, args, "value").unwrap().unwrap();
+
+        // Now, write the new value into memory.
+        // TODO: Validate it first?
+        match context
+            .platform
+            .settings_mut()
+            .set_json(key, value.as_bytes())
+        {
+            Ok(_) => match context.platform.save(context.buffer) {
+                Ok(_) => {
+                    writeln!(
+                            context,
```

Comment:
I'm a bit unhappy about writing to `context` vs writing to `context.platform.interface_mut()`. They are significantly different.

Reply:
Also we should check that the worst case write doesn't unduly starve the other things happening at idle prio (network, streaming IIRC).

Reply:
Indeed! It should not be using `write_all()`, but instead just `write()`. However, I believe writing to `context` instead of `interface_mut()` is correct here, as writing to `interface_mut()` will indefinitely block if `WriteReady::write_ready()` is false, which will essentially stall the USB port and starve the idle task.

Reply:
The problem is we don't write to `context` everywhere currently.

Reply:
Gotcha. I think this is a good reason why the interface should be explicitly owned by the `Context`, as it could then wrap the `Interface` in something that properly implements `core::fmt::Write`

Reply:
It doesn't seem appropriate to let this crate impl that trait.

Reply:
I fixed this in 2c391b669991052b5c450110bebcc7083ec989d4